### PR TITLE
[SPARK-14134] [core] [test-maven] Change the package name used for shading classes.

### PR DESCRIPTION
**Pull request by vanzin**
The current package name uses a dash, which is a little weird but seemed
to work. That is, until a new test tried to mock a class that references
one of those shaded types, and then things started failing.

Most changes are just noise to fix the logging configs.

For reference, SPARK-8815 also raised this issue, although at the time it
did not cause any issues in Spark, so it was not addressed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -178,6 +178,9 @@
     <test.java.home>${java.home}</test.java.home>
     <test.exclude.tags></test.exclude.tags>
 
+    <!-- Package to use when relocating shaded classes. -->
+    <spark.shade.packageName>org.spark_project</spark.shade.packageName>
```
Comment (srowen) [inline pom.xml]: For total avoidance of trouble, is `org.sparkproject` safer? or are you sure underscores are more OK than hyphens? BTW you could consider this PR as a resolution for SPARK-8815 too.

Comment (vanzin) [inline pom.xml]: Underscores are ok (they're not an operator like `-` is).

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
Weird error, seems unrelated. retest this please.

**Comment by vanzin** [comment]
hmm different test failure that also looks unrelated. retest this please

**Comment by vanzin** [comment]
sql test was stuck. retest this please.

**Comment by vanzin** [comment]
wtf. retest this please.

**Comment by vanzin** [comment]
this is the same flaky test that's failing in other places with maven... let me see if it's easy to fix.

**Comment by zsxwing** [comment]
@vanzio I fixed it in #11895

**Comment by zsxwing** [comment]
However, #11895 is blocked by #11940. It can not pass tests :(

**Comment by vanzin** [comment]
Cool, guess I'll wait for those, thanks.

**Comment by vanzin** [comment]
retest this please

**Comment by zsxwing** [comment]
You also need to update the flume's log4j files as it's moved back now.

**Comment by vanzin** [comment]
I'm gonna close this for now since #11796 is passing without this patch... no idea how.

**Comment by vanzin** [comment]
Reopened since I'm running into this problem locally, and the fix makes things more correct anyway.

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
@JoshRosen these tests are not happy with the build jdk; are you still in the process of fixing things?

**Comment by JoshRosen** [comment]
Argh, I forgot to do the `$PATH` prepending in the PRB... forget that those jobs' confs are managed via a different system than the rest of the Jenkins confs. Applying a manual conf. change to those jobs now.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by vanzin** [comment]
Since this was reviewed / tested before, I'll merge it in the morning unless someone objects (or beats me to it).

**Comment by JoshRosen** [comment]
LGTM, so I'm going to merge this now into 2.0. Thanks Marcelo!

**Pull request closed without merging**